torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -52,9 +55,10 @@ def fn(
             if type(input) in {torch.Tensor, features.Image} or isinstance(input, PIL.Image.Image):
                 return id, input
 
-            self._check_unsupported(input)
             return None
 
+        self._check_unsupported(sample)
```

Comment:
I think I should have been clearer here at https://github.com/pytorch/vision/pull/5500#discussion_r816514544. My fault.

Can we remove `_check_unsupported`, move the if statement in the `_extract_image` and parameterize `_extract_image` instead?